packages/react-devtools-shell/src/e2e/todo.js

```diff
@@ -0,0 +1,27 @@
+/** @flow */
+
+// This test harness mounts each test app as a separate root to test multi-root applications.
+
+import {createElement} from 'react';
+import {
+  // $FlowFixMe Flow does not yet know about createRoot()
+  createRoot,
+} from 'react-dom';
+import ToDoList from '../app/ToDoList';
+
+function createContainer() {
+  const container = document.createElement('div');
+
+  ((document.body: any): HTMLBodyElement).appendChild(container);
+
+  return container;
+}
+
+function mountApp(App) {
+  const container = createContainer();
+
+  const root = createRoot(container);
+  root.render(createElement(App));
+}
+
+mountApp(ToDoList);
```

Comment:
We might want to add e2e-specific things rather than reuse our ToDo app but this is okay for now.